### ISSUE TITLE
Feature request, support EC certificates

### ISSUE THREAD

**Issue by rarps** [usecallmanagernz/certutils#2]
In CUCM 12.5, support for EC certificates has come to the phones.  Currently, the tlvfile operation will only accept RSA certificates.  I've attempted to modify the script to handle EC certs, but I'm currently failing at that.

A side note, CUCM 12.5 also introduced the notion of an extended ITL file that a phone will request if the initial ITL file on 6970 has a signer of version 1.1.  I was able to update the script to handle the version attribute while parsing and building.

**Comment by gareth-palmer** [comment]
Adding support for EC certs should be easy enough, just need to reverse-engineer the format used to store the public key.

The CUCM 12.5 node I use for testing shows an extended ITL but the signer version is only 1.0, perhaps that is because it is not running in mixed-mode?

The only additional information I could find on extended ITLs was that they are only served via HTTPS which I guess would be on port 6971.

**Comment by rarps** [comment]
To set your signer to 1.1, go to system, enterprise parameters and enable Endpoint Advanced Encryption Algorithms Support.  The phone will then get the base ITL, switch to 6971 and get the extended ITL.

By the way, thank you for all the work you've put into this.  It's been immeasurably helpful!

**Comment by gareth-palmer** [comment]
Thanks for the information on how to enable extended ITL support.

I was able to guess the format of the public key (X962 format, uncompressed point) and support for EC certificates is now in the `master` branch.

The signer version can now be specified to `tlvfile` using the `-v` option, eg: `-v 1.1`. 

While enabling Advanced Endpoint Encryption I saw that SHA512 is now supported as a digest algorithm, so support for that has been added as well.
